Backend/Tumultu/Domain/Common/ValueObject.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+
+namespace Tumultu.Domain.Common;
+
+/// <summary>
+/// Value object created with Microsoft guidelines.
+/// Find more here: https://learn.microsoft.com/en-us/dotnet/architecture/microservices/microservice-ddd-cqrs-patterns/implement-value-objects
+/// </summary>
+public abstract class ValueObject
+{
+    protected abstract IEnumerable<object> GetEqualityComponents();
+    protected static bool EqualOperator(ValueObject l, ValueObject r)
+    {
+        if(l is null ^ r is null)
```

Comment:
very cool, can you add a comment that it's a XOR operator? Didn't know that and I'm certain that others might also not know that. Might save a search further down the line xd

Reply:
I thought it is kinda basic knowledge, but sure, I can add comment xD
This file is generally copypaste from MS docs so